packages/mui-system/src/cssVars/createCssVarsProvider.d.ts

```diff
@@ -2,73 +2,16 @@ import * as React from 'react';
 import getInitColorSchemeScript from './getInitColorSchemeScript';
 import { Mode, Result } from './useCurrentColorScheme';
 
-export type BuildCssVarsTheme<ThemeInput> = ThemeInput extends {
-  colorSchemes: Record<string, infer ColorSystems>;
-}
-  ? Omit<ThemeInput, 'colorSchemes'> &
-      ColorSystems & { vars: Omit<ThemeInput, 'colorSchemes'> & ColorSystems }
-  : never;
-
-/**
- * DesignSystemColorScheme: is what a design system provide by default. Mostly, `light` and `dark`
- * ApplicationColorScheme: is what developer can extend from a design system. Ex, `comfort` `trueDark` ...any name that they want
- *
- * This type enhance customization experience by checking if developer has extended the colorScheme or not (usually via module augmentation)
- * If yes, they must provide the palette of the extended colorScheme. Otherwise `theme` is optional.
- */
-type DecideTheme<
-  DesignSystemTheme extends { colorSchemes: Record<DesignSystemColorScheme, any> },
-  DesignSystemColorScheme extends string,
-  ApplicationTheme extends { colorSchemes: Record<ApplicationColorScheme, any> },
-  ApplicationColorScheme extends string | never,
-> = [ApplicationColorScheme] extends [never]
```

Comment:
Remove conditional type to simplify the code. The experience is likely the same as Material (all properties on theme input are optional)